lixPackageSets.lix_2_9{1,2}: drop

RaitoBezarius opened the pull request:
~~## Lix 2.90 series~~

~~Lix 2.90 is vulnerable to critical CVEs, known vulnerabilities warnings
were emitted. Users should have migrated by then.~~

~~There's no reason to keep it in-tree anymore.~~

This was already dropped in a previous PR.

## Lix 2.91 series

Lix 2.91 has been one of our historical LTS releases given that the Lix
2.92 series suffered from an unfortunate curl upstream bug and it took a
lot of time to fix it, in addition of other papercuts with SSH
connections.

Lix 2.91 is the last release not containing the asyncification work we
started.

Lix 2.92 latest minors sorted out these issues and possess bits of
asyncification, it's a much easier target for HEAD cherry-picks because
it also renamed the directory `src/nix` to `lix/`.

Plans are the following:

* Lix 2.92 is skipped.
* Lix 2.93 is the new stable.
* Lix 2.94 is the next stable.

Once NixOS 25.11 releases, Lix 2.92 will be dropped.

Lix 2.92 suffers from slight performance issues that are made up in Lix
2.93, your mileage may vary.

## Lix 2.92 series

Lix 2.92 is also dropped as we concluded that it is safe to skip over to 2.93 directly which we consider to be strongly stable. This simplifies maintenance for the Lix team.

If you are a user that discovers major regression in your workflows that worked in Lix 2.92, please contact or reach out the Lix team.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Add aliases for dropped packages
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

RaitoBezarius committed:
lixPackageSets.lix_2_91: drop

Lix 2.91 has been one of our historical LTS releases given that the Lix
2.92 series suffered from an unfortunate curl upstream bug and it took a
lot of time to fix it, in addition of other papercuts with SSH
connections.

Lix 2.91 is the last release not containing the asyncification work we
started.

Lix 2.92 latest minors sorted out these issues and possess bits of
asyncification, it's a much easier target for HEAD cherry-picks because
it also renamed the directory `src/nix` to `lix/`.

Plans are the following:

* Lix 2.92 is skipped.
* Lix 2.93 is the new stable.
* Lix 2.94 is the next stable.

Once NixOS 25.11 releases, Lix 2.92 will be dropped, therefore: it's
dropped from the HEAD as of now.

Lix 2.92 suffers from slight performance issues that are made up in Lix
2.93, your mileage may vary.

Change-Id: I83d0d1764edfc4009e4373f6b22d728390419e30
Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

RaitoBezarius commented on pkgs/tools/package-management/lix/default.nix:
```diff
@@ -262,10 +194,7 @@ lib.makeExtensible (self: {
-  # Note: This is not yet 2.92 because of a non-deterministic `curl` error.
-  # See: https://git.lix.systems/lix-project/lix/issues/662
-  stable = self.lix_2_91;
+  stable = self.lix_2_92;
```
Fine by me, in that case, we should probably consider pruning 2.92 as well.

RaitoBezarius replied:
This is done.

alois31 commented:
There are quite a few conditionals for older versions in [common-lix.nix](https://github.com/NixOS/nixpkgs/blob/c4ea7e484bd1de914d3b75481d6457bb06c6144e/pkgs/tools/package-management/lix/common-lix.nix) which could be removed along with the old versions. Similar for the nix-eval-jobs source which can always be assumed to be the same as lix if you end up dropping everything below 2.93.

emilazy commented:
(What macOS kernel bug, btw?)

RaitoBezarius commented:
Given that I have been in charge of the previous security issues, I think it's plausible we can do security until November but 2.91 is a highly expensive release, so November ought to be the utmost maximum support in nixpkgs IMHO.

RaitoBezarius committed:
lixPackageSets.lix_2_92: drop

After discussing with the nixpkgs developers and Lix developers, we came
to the conclusion that we could also drop directly 2.92 as this is not a
much used version and 2.93 has no known regression over 2.92 (only
advantages).

To reduce the maintenance churn on the Lix team, we drop this version
for 25.11 and will maintain only for the lifetime of 25.05.

Change-Id: I4ee310df31d36b92e394ff2dfebd589d10b12228
Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

RaitoBezarius commented:
This PR is now updated, hopefully, I understood the instructions given by the gentle stewards of this thread.

Please let me know if I missed a memo or something, I tried to update commit messages and body descriptions accordingly.

If I go silent for a while, it's that I'm (super)busy on some other things, feel free to take over my changes and proceed without me.

wolfgangwalther reviewed (approved):
Did not look at the diff, just confirming that I have no objections for these drops on master.

RaitoBezarius commented on pkgs/tools/package-management/lix/default.nix:
```diff
@@ -265,7 +185,6 @@ lib.makeExtensible (self: {
   };
 
   latest = self.lix_2_93;
-
```
should be dropped off the PR

RaitoBezarius commented on pkgs/tools/package-management/lix/default.nix:
```diff
@@ -279,7 +198,6 @@ lib.makeExtensible (self: {
           self.${version}.lix;
     in
     lib.dontRecurseIntoAttrs {
-      lix_2_91 = mkAlias "lix_2_91";
```
Dropping this made me wonder about whether we should add aliases-style information for users.
Any opinions from nixpkgs developers?
cc @emilazy

RaitoBezarius replied:
@emilazy in aliases.nix or in this file?

RaitoBezarius commented:
> Should we make 2.93 the default rather than 2.92? AIUI it fixes perf regressions that 2.92 had or something?

It's already the case in this PR.

emilazy reviewed (approved):
Sorry, I somehow missed that. LGTM, but aliases would be nice.

RaitoBezarius committed:
lix: add "removal messages" and restore old aliases

This ensures that evaluation doesn't suddenly break with cryptic
messages and add a nice message for users.

Change-Id: I1959ae7421c8a46ed93476bb014b84cfb26a4322
Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

wolfgangwalther reviewed (approved):
CI likes it\* and so do I.

\* except for `treefmt`, you'll need to run that once more ;)
